tests/visco_plastic_adiabatic_heating_density.prm

```diff
@@ -0,0 +1,113 @@
+# A test with the visco_plastic material model to ensure
+# that the multicomponent_incompressible EOS module
+# uses the adiabatic temperature to compute density when
+# adiabatic heating is turned on. The formula specified
+# in this test follows the Extended Boussinesq approximation.
+# The initial (constant) temperature throughout the model
+# has the same value as the Adiabatic surface temperature.
+# As a result, the density at the top (surface) of the model
+# should  be equal to the background value specified in the 
+# material model (3300):
+#   density = background_density * ( 1 - thermal_expansivity * (temperature - reference_temperature)),
+# where the reference_temperature is the adiabatic temperature.
+# At the base of the model the density should be lower than at the surface
+# due to the increase in adiabatic temperature with depth.
```

Comment:
I would add that normally we would use an adiabatic initial condition, and the density would then be constant with depth.